scipio/src/dma_file.rs

```diff
@@ -285,6 +285,48 @@ impl DmaFile {
         Reactor::get().alloc_dma_buffer(size)
     }
 
+    /// Returns the system's preferred alignment for the file
+    async fn fetch_device_alignment(&self) -> io::Result<u64> {
+        use std::fs;
+        use std::path;
+
+        // dev major + minor gives enough information about te file to determine physical location
+        let stats = self.statx().await?;
+        let major = stats.stx_dev_major;
+        let minor = stats.stx_dev_minor;
+
+        // /sys/dev/block/major:minor is a symlink to the device in /sys/devices/
+        let dir_path = format!("/sys/dev/block/{}:{}", major, minor);
+        let mut dir = path::Path::new(dir_path.as_str()).to_path_buf();
+        dir = dir.parent().unwrap().join(std::fs::read_link(&dir)?);
+
+        // if the device directory is a partition (contains a file named "partition"),
+        // we navigate to its parent directory, the host device
+        let device = if dir.join("/partition").exists() {
```

Comment:
This is ok but a bit wasteful or every open.

It would be better if we could parse sysfs entirely and return a struct that contains all its fields. It is totally fine if the only field we really parse is the one you need, since from that point on adding new fields becomes a mechanical task. For files that return an integer we can implement all of them by just passing their names though.

Once you have that you can store that structure in a hashmap by device id an avoid the parsing.

Long term I wanted to have an API that opens a file in a directory because that will allow us to use uring's registered fds. In that case we can do even better and store the result in the directory - not even needing the stat